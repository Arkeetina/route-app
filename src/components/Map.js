import React, { Component } from 'react';
import {
  withGoogleMap,
  GoogleMap,
  DirectionsRenderer,
} from 'react-google-maps';

class Map extends Component {
  constructor(props) {
    super(props);
    this.state = {
      directions: '',
    };
  }

  componentWillReceiveProps(nextProps) {
    const DirectionsService = new google.maps.DirectionsService();

    DirectionsService.route({
      // не круть, вообще не круть
      origin: new google.maps.LatLng(nextProps.path[0][0], nextProps.path[0][1]),
      waypoints: [{location: `${nextProps.path[1][0]}, ${nextProps.path[1][1]}`,}],
      destination: new google.maps.LatLng(nextProps.path[2][0], nextProps.path[2][1]),
      travelMode: google.maps.TravelMode.DRIVING,
    }, (result, status) => {
      if (status === google.maps.DirectionsStatus.OK) {
        this.setState({
          directions: result,
        });
      } else {
        console.error(`error fetching directions ${result}`);
      }
    });
  }

  render() {
    return (
      <GoogleMap
        defaultZoom={12}
        defaultCenter={new google.maps.LatLng(22.28552, 114.15769)}
      >
        {this.state.directions && <DirectionsRenderer directions={this.state.directions} />}
      </GoogleMap>
    );
  }
}

export default withGoogleMap(Map);
