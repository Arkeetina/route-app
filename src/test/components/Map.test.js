import React from 'react';
import { shallow } from 'enzyme';
import Map from '../../components/Map';

test('should render Map correctly', () => {
  const wrapper = shallow(
    <Map
      containerElement={<div style={{ height: `400px` }} />}
      mapElement={<div style={{ height: `100%` }} />}
    />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Map with path coordinates correctly', () => {
  const path = [
    ["22.372081", "114.107877"],
    ["22.326442", "114.167811"],
    ["22.284419", "114.159510"]
  ];
  const wrapper = shallow(
    <Map
      containerElement={<div style={{ height: `400px` }} />}
      mapElement={<div style={{ height: `100%` }} />}
      path={path}
    />);
  expect(wrapper).toMatchSnapshot();
});
