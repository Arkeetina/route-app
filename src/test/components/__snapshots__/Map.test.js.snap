// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Map correctly 1`] = `
<div
  style={
    Object {
      "height": "400px",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
  <div />
</div>
`;

exports[`should render Map with path coordinates correctly 1`] = `
<div
  style={
    Object {
      "height": "400px",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
  <div />
</div>
`;
